File: app/src/main/java/com/omar/retromp3recorder/app/di/ConfigModule.java
package com.omar.retromp3recorder.app.di;

import com.omar.retromp3recorder.app.recording.recorder.FilePathGenerator;
import com.omar.retromp3recorder.app.recording.recorder.RecorderDefaults;
import com.omar.retromp3recorder.app.recording.recorder.VoiceRecorder;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
class ConfigModule {
    @Singleton
    @Provides
    RecorderDefaults provideRecorderDefaults(FilePathGenerator filePathGenerator) {
        return new RecorderDefaults(
                VoiceRecorder.BitRate._320,
                VoiceRecorder.SampleRate._44100,
                filePathGenerator.generateFilePath()
        );
    }
}

File: app/src/main/java/com/omar/retromp3recorder/app/di/ConfigModule.kt
package com.omar.retromp3recorder.app.di

import com.omar.retromp3recorder.iorecorder.Mp3VoiceRecorder
import com.omar.retromp3recorder.iorecorder.RecorderDefaults
import dagger.Module
import dagger.Provides

@Module
class ConfigModule {
    @Provides
    fun provideRecorderDefaults(): RecorderDefaults {
        return RecorderDefaults(
            Mp3VoiceRecorder.BitRate._320,
            Mp3VoiceRecorder.SampleRate._44100
        )
    }

}
